src/errors/resolutionError.ts

```diff
@@ -79,7 +81,9 @@ export class ResolutionError extends Error {
   constructor(code: ResolutionErrorCode, options: ResolutionErrorOptions = {}) {
     const resolutionErrorHandler: ResolutionErrorHandler = HandlersByCode[code];
     const { domain, method, currencyTicker } = options;
-    super(resolutionErrorHandler(options));
+    let message = resolutionErrorHandler(options);
+    if (options.deprecated) {message += `\nResolutionErrorCode ${code} is deprecated and will be removed in the future`}
```

Comment:
When you specify deprecation, it is a good practice to tell people what to do instead.

```suggestion
    if (code === ResolutionErrorCode. UnspecifiedCurrency) {message += `\nResolutionErrorCode ${code} is deprecated and will be removed in the future. Use RecordNotFound code instead.`}
```

`deprecated` option is probably not needed in this case.